### PR TITLE
NOTICK- ECIES receives the salt as the lambda, allows for the flexibility to calculate the salt

### PR DESCRIPTION
**Pull request by alexey-kadyrov-r3**

### CODE REVIEW

File: components/crypto/crypto-service-impl/src/test/kotlin/net/corda/crypto/service/impl/CryptoOperationsTests.kt [hunk 1]
```diff
@@ -167,10 +168,12 @@ class CryptoOperationsTests {
                         signatureName = "RSA/NONE/PKCS1Padding",
                         customDigestName = DigestAlgorithmName(digest.algorithmName)
                     )
+
                     ECDSA_SECP256R1_CODE_NAME -> CustomSignatureSpec(
                         signatureName = "NONEwithECDSA",
                         customDigestName = DigestAlgorithmName(digest.algorithmName)
                     )
+
```
Comment (dickon) [inline components/crypto/crypto-service-impl/src/test/kotlin/net/corda/crypto/service/impl/CryptoOperationsTests.kt:176]: pointless whitespace change here and a few lines above, but not a merge blocker.

**Review by dickon — APPROVED** [review]
LGTM

File: components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EciesParams.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package net.corda.crypto.ecies
+
+/**
+ * Contains parameters used for HKDF
```
Comment (dimosr) [inline components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EciesParams.kt:4]: `Contains parameters used for HKDF` -> `Contains parameters used for ECIES`?

File: components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EciesParams.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ * @property salt used by the HKDF function and provides source of additional entropy, according to that spec
+ * https://datatracker.ietf.org/doc/html/rfc5869#section-3.1 it can be exchanged between
+ * communicating parties.
+ * @property aad the optional additional authentication data used by the GCM, the provided data will be concatenated
+ * with the public keys of the both parties.
```
Comment (dimosr) [inline components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EciesParams.kt:9]: `the provided data will be concatenated with the public keys of the both parties.` - is this the case? I thought that whatever is provided by the `EciesParamsProvider` is what is passed in as salt and add. And it's the responsibility of the implementation of the provider to concatenate whatever needed.

File: components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EncryptedDataWithKey.kt [hunk 4]
```diff
@@ -8,10 +8,13 @@ import java.security.PublicKey
  *
  * @param publicKey the public key of the generated ephemeral pair, note that the lifetime of the private key of the pair
  * and the shared secret generated using that key does not exceed the scope of the encrypting function.
+ * @param salt the salt which was used for deriving the shared secret.
+ * @param aad the aad which was used for deriving the shared secret.
```
Comment (dimosr) [inline components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EncryptedDataWithKey.kt:12]: `the aad which was used for deriving the shared secret.` -> `the aad which was provided in the AEAD encryption cipher`

File: components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EphemeralKeyPairEncryptor.kt [hunk 5]
```diff
@@ -22,28 +22,19 @@ interface EphemeralKeyPairEncryptor {
      * the keys and secrets. The digest algorithm for HKDF is infered based on the key scheme - e.g. SECP256R1 will
      * use SHA-256 and SECP384 will use SHA-384.
      *
-     * @param salt used by the HKDF function and provides source of additional entropy, according to that spec
-     * https://datatracker.ietf.org/doc/html/rfc5869#section-3.1 it can be exchanged between
-     * communicating parties
+     * @param params a [EciesParamsProvider] which used to get the salt and aad
```
Comment (dimosr) [inline components/crypto/crypto-ecies/src/main/kotlin/net/corda/crypto/ecies/EphemeralKeyPairEncryptor.kt:25]: nit: `which used to get the salt and aad` -> `which is used to get the salt and aad`

**Review by dimosr — COMMENTED** [review]
just got the chance to have a look at it, so posting a couple of comment/questions in case you want to have a look even though it's merged.

**Pull request merged**